contracts/mechanisms/LockDrop.sol

```diff
@@ -0,0 +1,183 @@
+pragma solidity ^0.4.24;
+
+import "openzeppelin-zos/contracts/math/SafeMath.sol";
+import "openzeppelin-zos/contracts/math/Math.sol";
+import "../token/EdgewareERC20.sol";
+
+/**
+ * The LockDrop contract locks up funds of depositors
+ * with a verifiable receipt/proof that can be used to
+ * redeem tokens on a separate, compatible blockchain.
+ * The tokens with be locked for a specified time length.
+ * 
+ * The tokens are minted at a rate that grows according
+ * to the chosen locking schedule.
+ */
+contract LockDrop {
+    EdgewareERC20 EDG;
+
+    uint public tokenCapacity;
+    uint public totalDeposits;
+    uint public totalEffectiveDeposits;
+    uint public initialValuation;
+    uint public globalPriceFloor;
+
+    uint public beginning;
+    uint public ending;
+
+    struct Lock {
+        uint amount;
+        uint effectiveAmount;
+        uint lockEnding;
+    }
+
+    mapping (address => Lock[]) locks;
+
+    event Deposit(address sender, uint value, uint effectiveValue, uint ending);
+    event Unlock(address sender, uint value);
+    event Withdraw(address sender, uint value, uint effectiveValue);
+
+    constructor(uint _lockPeriod, uint _tokenCapacity, uint _initialValuation, uint _priceFloor) public {
+        beginning = now;
+        ending = now + (1 days * _lockPeriod);
+        tokenCapacity = _tokenCapacity;
+        initialValuation = _initialValuation;
+        globalPriceFloor = _priceFloor;
+
+        EDG = new EdgewareERC20();
+    }
+
+    /**
+     * @dev        Lock function for participating in the lock drop
+     * @param      _length  The length of time chosen for locking ether
+     */
+    function lock(uint _length) payable public {
+        require( hasNotEnded() );
+        require( msg.value > 0 );
+        require( tokenCapacity > 0 );
+
+        uint dayLength = SafeMath.mul(_length, 1 days);
+        require( isValidLength(dayLength) );
+
+        uint discountAmount = discountedDepositValue(msg.value, dayLength);
+        totalDeposits = SafeMath.add(totalDeposits, msg.value);
+        totalEffectiveDeposits = SafeMath.add(totalEffectiveDeposits, effectiveAmount);
+
+        // TODO: Make sure price floor and initial valuation are valid units
+        uint price = Math.max256(globalPriceFloor, SafeMath.div(totalDeposits, initialValuation));
+        uint effectiveAmount = SafeMath.mul(discountAmount, price);
```

Comment:
Yea this should be SafeMath.div -> the price keeps going up, and as more tokens are distributed, each subsequent person should be getting less